backend/controllers/event.controller.ts

```diff
@@ -206,30 +222,34 @@ export const updateEvent = async (
 
     let updatedEvent: Event;
     if (hasPermission || isCreatedByUser) {
```

Comment:
This flow is a bit off. We are incorrectly throwing a permission error if updatedEvent doesn't exist.

This should work: 
```TypeScript
    if (!hasPermission && !isCreatedByUser) {
      throw new AppError(
        AppErrorName.PERMISSION_ERROR,
        `User does not have permission to update event`,
        403,
        true,
      );
    }

    if (req.file) {
      // update the file
      if (existingEvent?.image) {
        await deleteFromS3(existingEvent.image);
      }
      const uniqueFileName = generateUniqueFileName(
        req.file!.originalname,
        eventId,
      );
      const path = `images/events/${uniqueFileName}`;
      await UploadToS3(req.file!, path);
      validatedUpdateEventData.image = path;
    }

    // Update the event
    const updatedEvent = await prisma.event.update({
      where: { id: eventId },
      data: {
        ...validatedUpdateEventData,
      },
    });
    // send back the updated event
    if (updatedEvent) {
      // Event updated successfully
      res.status(200).json({
        message: "Event updated successfully",
        data: updatedEvent,
      });
    } else {
      throw new AppError(
        AppErrorName.EMPTY_RESULT_ERROR,
        "Event update returned empty result.",
        500,
        true,
      );
    }
  } catch (error: any) {
    next(error);
  }
};
```